settings/settings-cpt.php

```diff
@@ -159,19 +159,32 @@ protected function form() {
 	}
 
 	/**
-	 * Sanitizes the settings before saving.
+	 * Prepares the received data before saving.
 	 *
-	 * @since 1.8
+	 * @since 3.7
 	 *
-	 * @param array $options Unsanitized options to save.
+	 * @param array $options Raw values to save.
 	 * @return array
 	 */
-	protected function update( $options ) {
+	protected function prepare_raw_data( array $options ): array {
 		$newoptions = array();
 
 		foreach ( array( 'post_types', 'taxonomies' ) as $key ) {
 			$newoptions[ $key ] = empty( $options[ $key ] ) ? array() : array_keys( $options[ $key ], 1 );
 		}
+
+		// Limit to public post types.
+		$newoptions['post_types'] = array_intersect(
+			$newoptions['post_types'],
+			get_post_types( array( 'public' => true, '_builtin' => false ) )
+		);
+
+		// Limit to public taxonomies.
+		$newoptions['taxonomies'] = array_intersect(
+			$newoptions['taxonomies'],
+			get_taxonomies( array( 'public' => true, '_builtin' => false ) )
+		);
+
```

Comment:
I don't believe it's necessary. It's a choice to display only public types but the options can accept any types.

Reply:
Removed in [a6caec1](https://github.com/polylang/polylang/pull/1481/commits/a6caec13ed756fa831c107749755a95aa32982ad).